src/Cookie.php

```diff
@@ -37,91 +58,103 @@ final class Cookie
+     * @var bool|null whether the cookie should be accessible only through the HTTP protocol.
      * By setting this property to true, the cookie will not be accessible by scripting languages,
      * such as JavaScript, which can effectively help to reduce identity theft through XSS attacks.
      */
-    private bool $httpOnly = true;
+    private ?bool $httpOnly = null;
 
     /**
-     * @var string SameSite prevents the browser from sending this cookie along with cross-site requests.
-     * Please note that this feature is only supported since PHP 7.3.0
-     * For better security, an exception will be thrown if `sameSite` is set while using an unsupported version of PHP.
-     * To use this feature across different PHP versions check the version first. E.g.
-     * ```php
-     * $cookie->sameSite = PHP_VERSION_ID >= 70300 ? yii\web\Cookie::SAME_SITE_LAX : null,
-     * ```
-     * @see https://www.owasp.org/index.php/SameSite for more information about sameSite.
+     * @var string|null SameSite prevents the browser from sending this cookie along with cross-site requests.
+     * @see https://cheatsheetseries.owasp.org/cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.html#samesite-cookie-attribute for more information about sameSite.
      */
-    private string $sameSite = self::SAME_SITE_LAX;
+    private ?string $sameSite = null;
 
-    public function __construct(string $name, string $value)
+    public function __construct(string $name, string $value = '', bool $safeDefaults = true)
     {
-        // @see https://tools.ietf.org/html/rfc6265#section-4
-        // @see https://tools.ietf.org/html/rfc2616#section-2.2
-        if (!preg_match('~^[a-z0-9._\-]+$~i', $name)) {
-            throw new \InvalidArgumentException("The cookie name \"$name\" contains invalid characters.");
+        if (!preg_match(self::TOKEN, $name)) {
+            throw new InvalidArgumentException("The cookie name \"$name\" contains invalid characters.");
         }
 
         $this->name = $name;
-        $this->value = $value;
+        $this->setValue($value);
+
+        if ($safeDefaults) {
+            $this->path = '/';
+            $this->secure = true;
+            $this->httpOnly = true;
+            $this->sameSite = self::SAME_SITE_LAX;
+        }
     }
 
-    public function domain(string $domain): self
+    public function getName(): string
     {
-        $new = clone $this;
-        $new->domain = $domain;
-        return $new;
+        return $this->name;
     }
 
-    public function validFor(\DateInterval $dateInterval): self
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    public function expireAt(DateTimeInterface $dateTime): self
     {
-        $expireDateTime = (new \DateTimeImmutable())->add($dateInterval);
         $new = clone $this;
-        $new->expire = (int)$expireDateTime->format('U');
+        $new->expire = $dateTime->format('D, d M Y H:i:s T');
```

Comment:
Would be nice to extract format template as a private constant

Reply:
`\DateTimeInterface::RFC7231`

Reply:
@roxblnfk I first used that constant, but it produces different result than the one defined in RFC, the result of using that constant is e.g. `Sun, 06 Nov 1994 08:49:37 +0000` but according to [RFC](https://tools.ietf.org/html/rfc2616#section-3.3.1) we need `Sun, 06 Nov 1994 08:49:37 GMT`. I couldn't find any information in RFC whether the first format will be ok, so I've used the custom format.

@xepozz Will extract it to constant with link to rfc specification.

Reply:
@roxblnfk It occurs that `\DateTimeInterface::RFC7231` works correctly and tests just failed when I used it.

Reply:
RFC7231 is the preferred date and time format in modern HTTP headers